src/matlab2tikz.m

```diff
@@ -716,6 +716,40 @@ case guitypes()
     end
 end
 % ==============================================================================
+function [m2t, str] = addPlotyyReference(m2t, str, h)
+% Create labelled references to legend entries of the main plotyy axis
+
+% This ensures we are either on the main or secondary axis
+if ~isAxisPlotyy(m2t.currentHandles.gca), return, end
```

Comment:
Could you put that on separate lines. Otherwise one cannot put a break point on return.